rolling-shutter/mocksequencer/shutter.go

```diff
@@ -25,7 +27,43 @@ func (s *ShutterService) GetBatchIndex() (string, error) {
 	return hexutil.EncodeUint64(s.processor.batchIndex), nil
 }
 
-func (s *ShutterService) SubmitBatch(transaction string) (string, error) {
+func (s *ShutterService) validateBatch(tx *txtypes.Transaction) error {
+	if tx.Type() != txtypes.BatchTxType {
+		return errors.New("unexpected transaction type")
+	}
+	// TODO retrieve the collator for that block number (which could be higher then the last set
+	// "activation blocknumber")
+	if tx.Gas() > GasLimit {
+		return errors.New("gas limit reached")
+	}
+
+	// tx.Nonce() - ignore the nonce for now
+
+	if tx.ChainId().Cmp(s.processor.chainID) != 0 {
+		return errors.New("chain-id mismatch")
+	}
+
+	if s.processor.batchIndex != tx.BatchIndex()-1 {
+		return errors.New("incorrect batch-index for next batch")
+	}
+
+	collator, err := s.processor.collators.Find(tx.L1BlockNumber())
+	if err != nil {
+		return errors.Wrap(err, "collator validation failed")
+	}
+	sender, err := s.processor.signer.Sender(tx)
+	if err != nil {
+		return errors.Wrap(err, "error recovering batch tx sender")
+	}
+	if collator != sender {
+		return errors.Wrap(err, "not signed by correct collator")
+	}
+
+	// all checks passed, the batch-tx is valid
+	return nil
+}
+
+func (s *ShutterService) SubmitBatch(ctx context.Context, transaction string) (string, error) {
```

Comment:
I think it would be clearer to rename `transaction` to `batchTransaction` (similar to derived variables). Otherwise there's potential to confuse it with transactions included in the batch